src/checks/duplicated_keys.rs

```diff
@@ -0,0 +1,198 @@
+use crate::checks::{Check, Warning};
+use crate::extract_key;
+use crate::LineEntry;
+
+pub(crate) struct DuplicatedKeysChecker {
+    template: String,
+    keys: Vec<String>,
+}
+
+impl Default for DuplicatedKeysChecker {
+    fn default() -> Self {
+        Self {
+            keys: Vec::new(),
+            template: String::from("The {} key is duplicated"),
+        }
+    }
+}
+
+impl Check for DuplicatedKeysChecker {
+    fn run(&mut self, line: LineEntry) -> Option<Warning> {
+        let key = extract_key(&line.raw_string);
+
+        if self.keys.contains(&key) {
+            let warning = Warning::new(line, self.template.replace("{}", &key));
+            return Some(warning);
+        }
+
+        self.keys.push(key);
+        None
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn duplicated_keys_checker_run() {
+        let mut checker = DuplicatedKeysChecker::default();
+
+        let checks = vec![
+            (
+                LineEntry {
+                    number: 1,
+                    file_name: String::from(".env"),
+                    raw_string: String::from("RAILS_ENV=abc"),
+                },
+                None,
+            ),
+            (
+                LineEntry {
+                    number: 2,
+                    file_name: String::from(".env"),
+                    raw_string: String::from("RAILS_ENV=abc"),
+                },
+                Some(Warning::new(
+                    LineEntry {
+                        number: 2,
+                        file_name: String::from(".env"),
+                        raw_string: String::from("RAILS_ENV=abc"),
+                    },
+                    String::from("The RAILS_ENV key is duplicated"),
+                )),
+            ),
+        ];
+
+        for check in checks {
+            let (input, output) = check;
+            assert_eq!(checker.run(input), output);
+        }
+
+        // To clear the already saved keys
+        let mut checker = DuplicatedKeysChecker::default();
```

Comment:
Please divide these tests into several functions.
For example: https://github.com/mgrachev/dotenv-linter/blob/master/src/checks/incorrect_delimiter.rs#L33